pkg/storage/cache_test.go

```diff
@@ -22,17 +21,9 @@ func TestInMemoryCache(t *testing.T) {
 		require.Equal(t, "value", result)
 	})
 
-	t.Run("set_and_get_expired", func(t *testing.T) {
-		t.Cleanup(func() {
-			goleak.VerifyNone(t)
-		})
-		defer cache.Stop()
-		cache.Set("key", "value", -1*time.Nanosecond)
-		result := cache.Get("key")
-		require.Equal(t, "", result)
-	})
-
 	t.Run("stop_multiple_times", func(t *testing.T) {
```

Comment:
There seems to be a new implicit contract for `Stop` wherein the caller is allowed to call Stop multiple times (e.g. from diff goroutines). Should we add a test for this?

```
t.Run("stop_from_goroutines", func(t *testing.T) {
		cache, err := NewInMemoryLRUCache[string]()
		require.NoError(t, err)
		t.Cleanup(func() {
			goleak.VerifyNone(t)
		})

		pool := concurrency.NewPool(context.Background(), 2)
		pool.Go(func(ctx context.Context) error {
			cache.Stop()
			return nil
		})
		pool.Go(func(ctx context.Context) error {
			cache.Stop()
			return nil
		})
		err = pool.Wait()
		require.NoError(t, err)
	})
```